For #22147 - New search term groups telemetry

Mugurell opened the pull request:
3 new events added for when the user opens a tab, closes one or closes all from a search group.
Kimmy to confirm if the current https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/history_search_term_group_tapped is okay to be considered the 4th we needed.

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Mugurell commented:
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us understand how users interact with new search groups from history.
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * Better asses feature importance. 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
When a tab from the search term group is opened | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/22147
When a tab from the search term group is closed | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/22147
When all tabs from the search term group are closed | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/22147

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * One year (ending on November 1st, 2022), with the option to renew at that point. 
8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Amplitude. 
11. Where do you intend to share the results of your analysis?
  *  Only on Glean, Amplitude, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

jonalmeida commented on app/metrics.yaml:
```diff
@@ -2653,6 +2653,45 @@ history:
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/22147
+    data_reviews:
+      - ????????????
```
![raw](https://user-images.githubusercontent.com/1370580/140806694-1e0882de-fd6d-45aa-b71f-965feebf7c92.gif)

jonalmeida commented on app/metrics.yaml:
```diff
@@ -2653,6 +2653,45 @@ history:
+  search_term_group_open_tab:
+    type: event
+    description: |
+      A user opens a tab from the search term group.
```
From reading these descriptions, I thought these were probes for the tabs tray itself. Might be good to clarify the location where these probes are being added. For example..

```suggestion
      A user opens a tab from the search term group from the history list.
```

rocketsroger replied:
Agreed.  Looks like these are for History.  FYI, I recently add one telemetry for this requirement https://github.com/mozilla-mobile/fenix/pull/22300

rocketsroger reviewed (changes requested):
Please move these to the `history` category.  Also double check the PR I linked.  Thanks,

Mugurell commented:
> Please move these to the `history` category. Also double check the PR I linked. Thanks,

These new probes are already in the `history` category and use the naming scheme from [#22300](https://github.com/mozilla-mobile/fenix/pull/22300), them being recorded as
```
        {
          "timestamp": 10862,
          "category": "history",
          "name": "search_term_group_tapped"
        },
        {
          "timestamp": 14257,
          "category": "history",
          "name": "search_term_group_open_tab"
        },
                {
          "timestamp": 20389,
          "category": "history",
          "name": "search_term_group_remove_tab"
        },
        {
          "timestamp": 21755,
          "category": "history",
          "name": "search_term_group_remove_all"
        }
```
so in Glean similarly to https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/history_search_term_group_tapped the 3 events will actually be:
```
- history.search_term_group_open_tab
- history.search_term_group_remove_tab
- history.search_term_group_remove_all
```
Will add the history reference in the probe description and also add some tests.

rocketsroger commented:
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * This data will help us understand how users interact with new search groups from history.
> 
> 
>     1. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * Better asses feature importance.
> 
> 
>     1. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are no other alternatives.
> 
> 
>     1. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     1. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> When a tab from the search term group is opened 	Category 2 - interaction data 	#22147
> When a tab from the search term group is closed 	Category 2 - interaction data 	#22147
> When all tabs from the search term group are closed 	Category 2 - interaction data 	#22147
> 
>     1. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix%E2%80%A8)
> 
> 
>     1. How long will this data be collected? Choose one of the following:
> 
> 
>     * One year (ending on November 1st, 2022), with the option to renew at that point.
> 
> 
>     1. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     1. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     1. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Amplitude.
> 
> 
>     1. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean, Amplitude, and with mobile teams.
> 
> 
>     1. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by 2022-11-01

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

rocketsroger commented:
Looks good.  I've data reviewed.  Please update the data-review link in metrics.yml.  Thanks

Mugurell commented:
Thank you Roger!
Since I've also added some tests and updated that documentation I'll wait a bit for @jonalmeida to maybe have a second quick look, if everything seems okay.

Mugurell commented:
Kimmy said will also leave a comment on whether these events look right or not.
Adding the do-not-land label in the interim.

Mugurell commented:
Kimmy confirmed these events look good.

Mugurell commented:
Ohh I see @rocketsroger requested changes and the merge is blocked.
Asking for a re-review.
